2-Services/2.1-Special-Purpose-Services.bats

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bats
+
+@test "2.1.1.1 Ensure time synchronization is in use (Automated)" {
+		# Checks if one of the time synchronization tools is available
+		run bash -c "systemctl is-enabled systemd-timesyncd || dpkg -s chrony || dpkg -s ntp"
+		[[ "$status" -eq 0 ]]
+}
+
+@test "2.1.1.2 Ensure systemd-timesyncd is configured (Automated)" {
+		#It is recommended to use only one tool for time synchronization, be it ntp, chrony or the systemd-timesyncd service.
+		#If systemd-timesyncd is the chosen tool, this test verifies that systemd-timesyncd is configured correctly
+
+		run bash -c "dpkg -s ntp"
+		[[ "${lines[0]}" == "dpkg-query: package 'ntp' is not installed and no information is available" ]]
+
+		run bash -c "dpkg -s chrony"
+		[[ "${lines[0]}" == "dpkg-query: package 'chrony' is not installed and no information is available" ]]
+
+		run bash -c "systemctl is-enabled systemd-timesyncd.service"
+		[[ "$output" = "enabled" ]]
+
+
+		run bash -c "timedatectl status"
+		[[ "${lines[4]}" == *"System clock synchronized: yes"* ]]
+		[[ "${lines[5]}" == *"NTP service: active"* ]]
+}
+
+@test "2.1.1.3 Ensure chrony is configured (Automated)" {
+		#It is recommended to use only one tool for time synchronization, be it ntp, chrony or the systemd-timesyncd service.
+		#If chrony is the chosen tool, this test verifies that chrony is configured correctly
+
+		run bash -c "dpkg -s ntp | grep -E '(Status:|not installed)'"
+		[[ "${lines[0]}" == *"dpkg-query: package 'ntp' is not installed and no information is available"* ]]
+
```

Comment:
The check for systemd-timesyncd being masked is missing here

Reply:
If I understand correctly, you are referring to the following sentence: 
`Note: If chrony or ntp are used, systemd-timesyncd should be stopped and masked, and  this section skipped`.
I would add a test step in the steps that check the configuration of chrony and ntp to see if systemd-timesyncd is correctly set to "masked".

Reply:
There is one already defined in the CIS document.
```
Run the following command to verify that systemd-timsyncd is masked:
# systemctl is-enabled systemd-timesyncd
masked
```
And you already have a test in the 2.1.1.4 section for this:
```
run bash -c "systemctl is-enabled systemd-timesyncd.service"
[[ "$output" = "masked" ]]
```